cfgmgr/intfmgr.cpp

```diff
@@ -437,6 +455,12 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
             setIntfVrf(alias, vrf_name);
         }
 
+        /*Set the mac of interface*/
+        if (!mac.empty())
+        {
+            setIntfMac(alias, mac);
+        }
```

Comment:
if mac.empty() is true, that is the case of not having "mac_addr" field in CONFIG_DB. In this case, the convention is to push_back field-value tuple `"mac_addr": ""` to `data` to set to APPL_DB so that IntfsOrch does not make any assumption on default values. See the treatment on `mtu` and `adminStatus` on lines 493-497 and 499-504, respectively.

Reply:
Done

Reply:
Thank you for looking into the comment.

Just check the MacAddress class. When IntfOrch gets a value of empty string on line 424, it will use MacAddress::MacAddress(const std::string& macStr) to construct the MacAddress object. An empty string will cause MacAddress::parseMacString to return false, and further throw invalid_argument https://github.com/Azure/sonic-swss-common/blob/master/common/macaddress.cpp#L23

Did you try this change on a vs/a dut?

We may deem an empty string to be valid in constructing MacAddress, or put value "00:00:00:00:00:00" which is deem valid in current parseMacString logic.

Reply:
Yes, In fact I exactly put "00:00:00:00:00:00" string in app-db